warehouse/router/state_generate_upload_schema.go

```diff
@@ -0,0 +1,35 @@
+package router
+
+import (
+	"encoding/json"
+	"fmt"
+
+	"github.com/rudderlabs/rudder-server/warehouse/internal/repo"
+)
+
+func (job *UploadJob) generateUploadSchema() error {
+	uploadSchema, err := job.schemaHandle.ConsolidateStagingFilesUsingLocalSchema(job.ctx, job.stagingFiles)
+	if err != nil {
+		return fmt.Errorf("consolidate staging files schema using warehouse schema: %w", err)
+	}
+
+	marshalledSchema, err := json.Marshal(uploadSchema)
+	if err != nil {
+		return err
```

Comment:
Can we add some context here?

Reply:
Is it okay, if I take this up in a follow-up PR after this PR gets merged? I wanted this standalone PR to just split the states, and later we can work on individual states.